### ISSUE TITLE
Running SGE in fp16 (Mixed Precision)

### ISSUE THREAD

**Issue by hussam789** [implus/PytorchInsight#32]
Hi,
Running SGE Module in mixed precision resulted in gradient overflow (NaN in training loss).
I converted it the input to float and then to half, that helped a bit at first but half way through the training i still got Nan in training loss.

Did anyone test SGE with mixed precision?

**Comment by implus** [comment]
I have a suggestion: try to change the form of "sqrt(xxx) + eps" into "sqrt(xxx + eps)" in SGE module. 
In the repo, it becomes "t.std(dim=1, keepdim=True) + 1e-5" -> "torch.sqrt(t.var() + 1e-5)". You can refer to https://github.com/joe-siyuan-qiao/WeightStandardization/issues/16 for more details. If this solution can not help, it is probably due to the mixed precision problem.

**Comment by mrT23** [comment]
one more suggestion for numerical stability-
try to combine the std division with the weight multiplication:

instead of:
```
t = t / std
t = t.view(b, self.groups, h, w)
t = t * self.weight + self.bias
```

do:
```
std = std.view(b, self.groups)
t = t.view(b, self.groups, h, w)
t = ((t * self.weight) / std[:, :, None, None] + self.bias)
```
